### PR TITLE
tdd

### PR DESCRIPTION
**Pull request by Naledi-Dikgale**
### In this Pull Request we did the following

- [x] Created a class called `Solver` with a method called factorial that takes one argument an integer and returns the factorial for that number.

- [x] Created a method called `reverse` that takes one argument, a string word, and returns a word reversed.

- [x] Created a method called `fizzbuzz`  that takes one argument, an integer N, and returns a string.

- [x] Any other case, return N as a string.

### CODE REVIEW

File: solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  describe '#factorial' do
+    it 'returns 1 when given 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'returns 1 when given 1' do
+      expect(solver.factorial(1)).to eq(1)
+    end
+
```
Comment (omarsalem7) [inline solver_spec.rb:13]: - [ ] Kindly consider adding a test case when the given negative value which raises an exception

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi team😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights

- Solver class created properly ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by osamaashraf6 — APPROVED** [review]
Hi @Naledi-Dikgale 😊, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
MinionsGIF (3)

## Highlights🎉🎉

- [ ] All tests passed💯
- [ ] The reverse method has tests that handle👍
- [ ] The `#fizzbuzz` method has tests that handle🚀
- [ ] The `#factorial` method has tests that handle

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
